src/ast-parser.y

```diff
@@ -1324,66 +1363,68 @@ script_var_opt :
 
 action :
     LPAR INVOKE script_var_opt quoted_text const_list RPAR {
-      WABT_ZERO_MEMORY($$);
-      $$.loc = @2;
-      $$.module_var = $3;
-      $$.type = ActionType::Invoke;
-      $$.invoke.name = $4;
-      $$.invoke.args = $5;
+      $$ = new Action();
```

Comment:
Again, you might consider some explicit constructors for Action, to match the cases of the parser. This could also be applied to most parsing constructs.

Note: This doesn't have to be in this PR, just a suggestion on further cleanup.